src/framework/uicomponents/qml/Muse/UiComponents/internal/StyledDropdownView.qml

```diff
@@ -55,6 +55,10 @@ DropdownView {
     property int contentWidth: root.itemWidth
     property int contentHeight: content.contentBodyHeight
 
+    property string typeAheadStr: ""
+    property string lastTypedChar: ""
```

Comment:
It's not quite enough to check only the last typed character. You need to make sure that the entire string is just one repeated character. For example `oo` counts as a repeat but `boo` shouldn't count as a repeat.

That being said, the characters are typed one at a time, and there's no point checking characters that have already been checked. So what you actually need is this:

```QML
property string typeAheadStr: ""
property string typeAheadLastChar: ""
property bool typeAheadSameChar: true

property Timer typeAheadTimer: Timer {
    interval: 1000
    repeat: false
    onTriggered: {
        typeAheadStr = ""
        typeAheadLastChar = ""
        typeAheadSameChar = true
    }
}
```

When each typed character arrives in `Keys.onShortcutOverride`, you'll check to see if it matches the previous one. If it doesn't match then you'll set `typeAheadSameChar` to `false`. Once it has been set to false it should remain that way until the timer runs out, even if subsequently typed characters do happen to match.

---

The `typeAheadLastChar` variable isn't strictly necessary since you could just check the final character in `typeAheadStr`. However, in some languages (e.g. Chinese) it's possible you might get a 'glyph' that is more than one 'character' in length, so for this reason I think it's safest to keep the `typeAheadLastChar` variable.